src/Service/StatisticsExtractionService.php

```diff
@@ -218,13 +222,22 @@ public function removeExtractedEntries(\DateTime $compareDate)
     {
         $entries = $this->entryRepository->findBy(['extracted' => true]);
 
+        $entriesRemoved = 0;
+
         /* @var Entry $entry */
         foreach ($entries as $entry) {
             if (null !== $entry->getExtracted() && null !== $entry->getExtractionDate()) {
                 $diff = (int) $entry->getExtractionDate()->diff($compareDate)->format('%a');
 
                 if (0 < $diff && $entry->getExtractionDate() < $compareDate) {
                     $this->documentManager->remove($entry);
+
+                    $entriesRemoved++;
+
+                    // Flush when batch size is exceeded to avoid memory buildup.
+                    if ($entriesRemoved % self::BATCH_SIZE === 0) {
+                        $this->documentManager->flush();
```

Comment:
What happens if the removed size is smaller than the batch size or the last batch is smaller?

Reply:
Then the loop terminates and
$this->documentManager->flush();
is called at the end.

Reply:
https://github.com/danskernesdigitalebibliotek/ddb-cover-service-faktor-export/pull/1/files/71aaceadc8606265f89225c4b92287472f9840c7..cc2e5532946b705f694ff4b76a1aa2ad1aacfcf1#diff-e8ce8b090fc1981cbba9e4709b0673f4R245